validator_client/src/validator_store.rs

```diff
@@ -154,15 +219,39 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
         validator_pubkey: &PublicKey,
         mut block: BeaconBlock<E>,
     ) -> Option<BeaconBlock<E>> {
-        // TODO: check for slashing.
-        self.validators
-            .read()
-            .get(validator_pubkey)
-            .and_then(|validator_dir| {
-                let voting_keypair = validator_dir.voting_keypair.as_ref()?;
-                block.sign(&voting_keypair.sk, &self.fork()?, &self.spec);
-                Some(block)
-            })
+        let validators = self.validators.read();
+
+        // Retrieving the corresponding ValidatorDir
+        let validator = match validators.get(validator_pubkey) {
+            Some(validator) => validator,
+            None => return None,
```

Comment:
Maybe we should log something here, stating that the validator_pubkey doesn't appear in amongst the validators?